src/cc/export/helpers.h

```diff
@@ -1216,6 +1221,18 @@ int bpf_usdt_readarg_p(int argc, struct pt_regs *ctx, void *buf, u64 len) asm("l
 #define PT_REGS_RC(x) ((x)->regs[2])
 #define PT_REGS_SP(x) ((x)->regs[29])
 #define PT_REGS_IP(x) ((x)->cp0_epc)
+#elif defined(bpf_target_loongarch)
+#define PT_REGS_PARM1(x) ((x)->regs[4])
+#define PT_REGS_PARM2(x) ((x)->regs[5])
+#define PT_REGS_PARM3(x) ((x)->regs[6])
+#define PT_REGS_PARM4(x) ((x)->regs[7])
+#define PT_REGS_PARM5(x) ((x)->regs[8])
+#define PT_REGS_PARM6(x) ((x)->regs[9])
+#define PT_REGS_RET(x) ((x)->regs[1])
+#define PT_REGS_FP(x) ((x)->regs[22]) /* Works only with CONFIG_FRAME_POINTER */
+#define PT_REGS_RC(x) ((x)->regs[4])
+#define PT_REGS_SP(x) ((x)->regs[3])
+#define PT_REGS_IP(x) ((x)->csr_epc)
```

Comment:
These bits depend on the kernel ABI being stable. We might have to wait until the LoongArch port of Linux gets upstreamed or at least settles down, only then this could be merged without code churn.

Reply:
Thank you for your review. We actually have an ABI register conversion as below, it is settled for a long time.

```
================= =============== =================== ============
Name              Alias           Usage               Preserved
                                                      across calls
================= =============== =================== ============
``$r0``           ``$zero``       Constant zero       Unused
``$r1``           ``$ra``         Return address      No
``$r2``           ``$tp``         TLS                 Unused
``$r3``           ``$sp``         Stack pointer       Yes
``$r4``-``$r11``  ``$a0``-``$a7`` Argument registers  No
``$r4``-``$r5``   ``$v0``-``$v1`` Return value        No
``$r12``-``$r20`` ``$t0``-``$t8`` Temp registers      No
``$r21``          ``$x``          Reserved            Unused
``$r22``          ``$fp``         Frame pointer       Yes
``$r23``-``$r31`` ``$s0``-``$s8`` Static registers    Yes
================= =============== =================== ============
```
https://git.kernel.org/pub/scm/linux/kernel/git/chenhuacai/linux-loongson.git/tree/Documentation/loongarch/introduction.rst?h=loongarch-next

Reply:
> Thank you for your review. We actually have an ABI register conversion as below, it is settled for a long time.
> 
> ```
> ================= =============== =================== ============
> Name              Alias           Usage               Preserved
>                                                       across calls
> ================= =============== =================== ============
> ``$r0``           ``$zero``       Constant zero       Unused
> ``$r1``           ``$ra``         Return address      No
> ``$r2``           ``$tp``         TLS                 Unused
> ``$r3``           ``$sp``         Stack pointer       Yes
> ``$r4``-``$r11``  ``$a0``-``$a7`` Argument registers  No
> ``$r4``-``$r5``   ``$v0``-``$v1`` Return value        No
> ``$r12``-``$r20`` ``$t0``-``$t8`` Temp registers      No
> ``$r21``          ``$x``          Reserved            Unused
> ``$r22``          ``$fp``         Frame pointer       Yes
> ``$r23``-``$r31`` ``$s0``-``$s8`` Static registers    Yes
> ================= =============== =================== ============
> ```
> https://git.kernel.org/pub/scm/linux/kernel/git/chenhuacai/linux-loongson.git/tree/Documentation/loongarch/introduction.rst?h=loongarch-next

Hi, what I mean by "ABI" is not limited to register convention; things like the field name `regs` also count.